solr/core/src/java/org/apache/solr/util/EnvUtils.java

```diff
@@ -0,0 +1,279 @@
+  public static String getEnv(String key) {
+    return ENV.get(key);
+  }
+
+  /** Get a single environment variable as string, or default */
+  public static String getEnv(String key, String defaultValue) {
+    return ENV.getOrDefault(key, defaultValue);
+  }
+
+  /** Get an environment variable as long */
+  public static long getEnvAsLong(String key) {
+    return Long.parseLong(ENV.get(key));
+  }
+
+  /** Get an environment variable as long, or default value */
+  public static long getEnvAsLong(String key, long defaultValue) {
+    String value = ENV.get(key);
+    if (value == null) {
+      return defaultValue;
+    }
+    return Long.parseLong(value);
+  }
+
+  /** Get an env var as boolean */
+  public static boolean getEnvAsBool(String key) {
+    return StrUtils.parseBool(ENV.get(key));
+  }
+
+  /** Get an env var as boolean, or default value */
+  public static boolean getEnvAsBool(String key, boolean defaultValue) {
+    String value = ENV.get(key);
+    if (value == null) {
+      return defaultValue;
+    }
+    return StrUtils.parseBool(value);
+  }
+
+  /** Get comma separated strings from env as List */
+  public static List<String> getEnvAsList(String key) {
+    return getEnv(key) != null ? stringValueToList(getEnv(key)) : null;
+  }
+
+  /** Get comma separated strings from env as List */
+  public static List<String> getEnvAsList(String key, List<String> defaultValue) {
+    return ENV.get(key) != null ? getEnvAsList(key) : defaultValue;
+  }
+
+  /** Set an environment variable */
+  public static void setEnv(String key, String value) {
+    ENV.put(key, value);
+  }
+
+  /** Set all environment variables */
+  public static synchronized void setEnvs(Map<String, String> env) {
+    ENV.clear();
+    ENV.putAll(env);
+  }
+
+  /** Get all Solr system properties as a sorted map */
+  public static SortedMap<String, String> getProps() {
+    ensureInitialized();
+    SortedMap<String, String> props = new TreeMap<>();
+    for (String key : System.getProperties().stringPropertyNames()) {
+      props.put(key, System.getProperty(key));
+    }
+    return props;
+  }
+
+  /** Get a property as string */
+  public static String getProp(String key) {
+    return getProp(key, null);
+  }
+
+  /**
+   * Get a property as string with a fallback value. All other getProp* methods use this.
+   *
+   * @param key property key, which treats 'camelCase' the same as 'camel.case'
+   */
+  public static String getProp(String key, String defaultValue) {
+    ensureInitialized(); // Avoid race condition with init()
+    String value = getPropWithCamelCaseFallback(key);
+    return value != null ? value : defaultValue;
+  }
+
+  /**
+   * Get a property from given key or an alias key converted from CamelCase to dot separated.
+   *
+   * @return property value or value of dot-separated alias key or null if not found
+   */
+  private static String getPropWithCamelCaseFallback(String key) {
```

Comment:
We should put cached answers in a ConcurrentHashMap so that string manipulations needn't be re-parsed/resolved on every fetch (calling camelCaseToDotSeparated).  The query path might reasonably want to call getProp and it'd be annoying to see that for obscure features.